pkg/builder/kaniko/git.go

```diff
@@ -0,0 +1,47 @@
+package kaniko
+
+import (
+	"regexp"
+	"strings"
+)
+
+type GitContext struct {
+	Repo     string
+	Commit   string
+	Username string
+	Password string
+}
+
+func GetGitBuildContext(g *GitContext) (string, error) {
```

Comment:
feel like this should be defined on GitContext, vs passed in, so maybe it reads more like

```go

gc := GitContext{...} // fill with stuff
gc.AsContextString() 
// OR
gc.BuildContext() 
```

etc, would look nice...

Reply:
good point, thanks :)